src/sentry/integrations/msteams/card_builder/__init__.py

```diff
@@ -137,128 +149,105 @@ def build_group_footer(group: Group, rules: Sequence[Rule], event: Event) -> Col
-                            {"title": "1 time", "value": 1},
-                            {"title": "10 times", "value": 10},
-                            {"title": "100 times", "value": 100},
-                            {"title": "1,000 times", "value": 1000},
-                            {"title": "10,000 times", "value": 10000},
-                        ],
-                    },
-                ],
-                "actions": [
-                    {
-                        "type": "Action.Submit",
-                        "title": "Ignore",
-                        "data": generate_action_payload(
-                            ACTION_TYPE.IGNORE, event, rules, integration
-                        ),
-                    }
-                ],
-            },
-        }
+    ignore_action = create_issue_action_block(
+        event=event,
+        rules=rules,
+        integration=integration,
+        toggled=GroupStatus.IGNORED == status,
+        action=ACTION_TYPE.IGNORE,
+        action_title=IssueConstants.IGNORE,
+        reverse_action=ACTION_TYPE.UNRESOLVE,
+        reverse_action_title=IssueConstants.STOP_IGNORING,
+        # card_kwargs
+        title=IssueConstants.IGNORE_INPUT_TITLE,
+        input_id=IssueConstants.IGNORE_INPUT_ID,
+        choices=IssueConstants.IGNORE_INPUT_CHOICES,
+    )
 
-    if group.get_assignee():
-        assign_action = {
-            "type": "Action.Submit",
-            "title": "Unassign",
-            "data": generate_action_payload(ACTION_TYPE.UNASSIGN, event, rules, integration),
-        }
-    else:
-        teams_list = group.project.teams.all().order_by("slug")
-        teams = [{"title": f"#{u.slug}", "value": f"team:{u.id}"} for u in teams_list]
-        teams = [{"title": "Me", "value": ME}] + teams
-        assign_action = {
-            "type": "Action.ShowCard",
-            "version": "1.2",
-            "title": "Assign",
-            "card": {
-                "type": "AdaptiveCard",
-                "body": [
-                    {"type": "Input.ChoiceSet", "id": "assignInput", "value": ME, "choices": teams}
-                ],
-                "actions": [
-                    {
-                        "type": "Action.Submit",
-                        "title": "Assign",
-                        "data": generate_action_payload(
-                            ACTION_TYPE.ASSIGN, event, rules, integration
-                        ),
-                    }
-                ],
-            },
-        }
+    teams_choices = get_teams_choices(group)
+
+    assign_action = create_issue_action_block(
+        event=event,
+        rules=rules,
+        integration=integration,
+        toggled=group.get_assignee(),
+        action=ACTION_TYPE.ASSIGN,
+        action_title=IssueConstants.ASSIGN,
+        reverse_action=ACTION_TYPE.UNASSIGN,
+        reverse_action_title=IssueConstants.UNASSIGN,
+        # card_kwargs
+        title=IssueConstants.ASSIGN_INPUT_TITLE,
```

Comment:
Added an extra title which is not present in the original card to make all of the actions consistent.